File: SampleProject/SampleProject/ThirdViewController.m
//
//  ThirdViewController.m
//  SampleProject
//
//  Created by Pollfish on 12/05/2019.
//  Copyright © 2019 POLLFISH. All rights reserved.
//


#import "ThirdViewController.h"

@interface ThirdViewController()

@end

@implementation ThirdViewController

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    if (@available(iOS 14, *)) {
        [self requestIDFAPermission];
    } else {
        [self initPollfish];
    }
}

- (void)requestIDFAPermission {
#if __has_include(<AppTrackingTransparency/AppTrackingTransparency.h>)
    [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
      dispatch_async(dispatch_get_main_queue(), ^{
          if (status == ATTrackingManagerAuthorizationStatusAuthorized) {
              [self initPollfish];
          } else {
              [self showNoPermissionAlert];
          }
      });
    }];
#endif
}

- (void) initPollfish{
    
    NSLog(@"Pollfish: init");
    
    PollfishParams *pollfishParams = [[PollfishParams alloc] init:@"af89aaf1-b7d4-46c1-8e91-b2625c2d5dbe"];
    [pollfishParams releaseMode:false];
    [pollfishParams offerwallMode:true];
    [pollfishParams rewardMode:true];
    [pollfishParams requestUUID:@"my_id"];
    
    [Pollfish initWith:pollfishParams delegate:self];
    
    _offerwallBtn.hidden=true;
    _loggingLabel.text=@"Logging area..";
}

- (void)pollfishSurveyReceivedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    NSLog(@"Pollfish: Survey Received - Offerwall available");
    
    _offerwallBtn.hidden=false;
    _loggingLabel.text=[NSString stringWithFormat:@"Pollfish: Survey Received - Offerwall available"];
}


- (void) pollfishSurveyCompletedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    int surveyPrice = [[surveyInfo cpa] intValue];
    int surveyIR = [[surveyInfo ir] intValue];
    int surveyLOI = [[surveyInfo loi] intValue];
    
    NSString *surveyClass = [surveyInfo surveyClass];
    
    NSString *rewardName = [surveyInfo rewardName];
    int rewardValue = [[surveyInfo rewardValue] intValue];
    
    NSLog(@"Pollfish: Survey Completed - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue);
    
    _loggingLabel.text=[NSString stringWithFormat:@"Pollfish: Survey Completed - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue];

    // in a real world app you should wait for s2s callbacks prior rewarding your user
}

- (void)pollfishOpened
{
    NSLog(@"Pollfish: Survey Opened");
    
    _loggingLabel.text=@"Pollfish: Survey Opened";
}

- (void)pollfishClosed
{
    NSLog(@"Pollfish: Survey Closed");
}

- (void)pollfishNotAvailable
{
    NSLog(@"Pollfish: Offerwall Not Available");
    
    _loggingLabel.text=@"Pollfish: Offerwall Not Available";
}

- (void)pollfishUsernotEligible
{
    NSLog(@"Pollfish: User Not Eligible");
    
    _loggingLabel.text=@"Pollfish: User Not Eligible";
}

-(IBAction)showPollfish:(id)sender{
    [Pollfish show];
}

- (void) pollfishUserRejectedSurvey
{
    NSLog(@"Pollfish: User Rejected Survey");
    
    _loggingLabel.text=@"Pollfish: User Rejected Survey";
    _offerwallBtn.hidden=true;
}

@end

File: SampleProjectSwift/SampleProjectSwift/ThirdViewController.swift
//
//  ThirdViewController.swift
//  SampleProjectSwift
//
//  Created by Pollfish on 13/05/2019.
//  Copyright © 2019 POLLFISH. All rights reserved.
//


import UIKit
import Pollfish
#if canImport(AppTrackingTransparency)
import AppTrackingTransparency
#endif

class ThirdViewController: UIViewController, PollfishDelegate {
    
    
    @IBOutlet weak var offerwallBtn: UIButton!
    @IBOutlet weak var loggingLabel: UILabel!
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
                
        loggingLabel.text="Logging area.."
        offerwallBtn.isHidden=true
        
        if #available(iOS 14, *) {
            requestIDFAPermission()
        } else {
            initPollfish()
        }
        
        loggingLabel.text="Logging area.."
    }
    
    @available(iOS 14, *)
    func requestIDFAPermission() {
        #if canImport(AppTrackingTransparency)
        ATTrackingManager.requestTrackingAuthorization { status in
            DispatchQueue.main.async {
                switch status {
                case .authorized:
                    self.initPollfish()
                default:
                    self.showNoPermissionAlert()
                }
            }
        }
        #endif
    }
    
    func initPollfish() {
        let pollfishParams = PollfishParams("YOUR_API_KEY")
            .rewardMode(true)
            .offerwallMode(true)
            .releaseMode(false)
            .requestUUID("my_id")
        
        Pollfish.initWith(pollfishParams, delegate: self)
    }
    
    func pollfishSurveyNotAvailable() {
        print("pollfishNotAvailable")
        
        loggingLabel.text="Pollfish - Survey Not Available"
    }
    
    func pollfishSurveyReceived(surveyInfo: SurveyInfo?) {
        print("Pollfish Survey Received")
        
        loggingLabel.text="Pollfish Survey Received"
        offerwallBtn.isHidden=false
    }
    
    func pollfishOpened() {
        print("pollfishOpened")
        
        loggingLabel.text="Pollfish - Pollfish Panel Opened"
    }
    
    func pollfishClosed() {
        print("pollfishClosed")
    }
    
    func pollfishUserNotEligible() {
        print("pollfishUsernotEligible")
        
        loggingLabel.text="Pollfish - User Not Eligible"
    }
    
    func pollfishSurveyCompleted(surveyInfo: SurveyInfo) {
        let surveyPrice = surveyInfo.cpa?.intValue
        let surveyIR = surveyInfo.ir?.intValue
        let surveyLOI = surveyInfo.loi?.intValue
        let surveyClass = surveyInfo.surveyClass
        
        let rewardName = surveyInfo.rewardName
        let rewardValue = surveyInfo.rewardValue?.intValue
        
        print("Pollfish Survey  Completed - SurveyPrice: \(String(describing: surveyPrice)) andSurveyIR: \(String(describing: surveyIR)) andSurveyLOI: \(String(describing: surveyLOI)) andSurveyClass: \(String(describing: surveyClass)) andRewardName: \(String(describing: rewardName)) andRewardValue:\(String(describing: rewardValue))")
        
        loggingLabel.text="Pollfish Survey Completed - You won \(rewardValue ?? 0) \(rewardName ?? "")"
        
        // in a real world app you should wait for s2s callbacks prior rewarding your user
    }

    @IBAction func showPollfish(_ sender: AnyObject) {
        print("showPollfish")
        
        Pollfish.show()
    }
    
    @IBAction func hidePollfish(_ sender: AnyObject) {
        print("hidePollfish")
        
        Pollfish.hide()
    }
    
}
